scrapers_next/mo/committees.py

```diff
@@ -0,0 +1,187 @@
+
+
+class SenateCommitteeList(HtmlListPage):
+    source = "https://www.senate.mo.gov/committee/"
+    selector = CSS("#post-90 a")
+
+    def process_item(self, item):
+
+        type = item.text_content()
+        if type == "Standing" or type == "Statutory":
+            return SenateTypeCommitteeList(source=item.get("href"))
+        else:
+            self.skip()
+
+
+class HouseCommitteeDetail(HtmlPage):
+    # example_source = "https://www.house.mo.gov/MemberGridCluster.aspx?filter=compage&category=committee&Committees.aspx?category=all&committee=2582&year=2021&code=R"
+    example_source = "https://www.house.mo.gov/MemberGridCluster.aspx?filter=compage&category=committee&Committees.aspx?category=all&committee=2571&year=2021&code=R"
+
+    def process_page(self):
+        com = self.input
+        com.add_source(self.source.url)
+        link = self.source.url.replace(
+            "MemberGridCluster.aspx?filter=compage&category=committee&", ""
+        )
+        com.add_link(link, note="homepage")
+
+        # As of now, one committees page is empty. Just in case it is updated soon, the page will still be scraped
+        try:
+            members = CSS("#theTable tr").match(self.root)
+
+            for member in members:
+                # skip first row with table headers
+                if member.text_content().strip() == "NamePartyPosition":
+                    continue
+
+                __, name, __, position = CSS("td").match(member)
+
+                name = (
+                    name.text_content()
+                    .replace("Rep. ", "")
+                    .replace("Sen. ", "")
+                    .split(",")
+                )
+                name = name[1] + " " + name[0]
+
+                if position.text_content():
+                    position = position.text_content()
+                else:
+                    position = "member"
+
+                com.add_member(name, position)
+        except SelectorError:
+            print("selector error")
+            pass
+        return com
+
+
+class HouseCommitteeList(HtmlListPage):
+    source = "https://www.house.mo.gov/CommitteeHierarchy.aspx"
+    selector = CSS("a")
+    chamber = "lower"
+
+    def process_item(self, item):
+        committee_name = item.text_content()
+
+        committee_name = (
+            committee_name.replace("Joint Committee on the", "")
+            .replace("Joint Committee on", "")
+            .replace("Special Committee on", "")
+            .replace("Special Interim Committee on", "")
+            .replace(", Standing", "")
+            .replace(", Statutory", "")
+            .replace(", Interim", "")
+            .replace(", Special Standing", "")
+        )
+        committee_name = committee_name.strip()
+
+        if "Subcommittee" in committee_name:
+            name = committee_name.replace("Subcommittee on ", "").replace(
+                ", Subcommittee", ""
+            )
+
+            parent = (
```

Comment:
code is same here

Reply:
I think I have to keep this here so that I don't replace the "Subcommittee" in the original string before I check for it and then replace it